packages/keyring-controller/src/KeyringController.ts

```diff
@@ -1831,29 +1910,35 @@ export class KeyringController extends BaseController<
     this.#password = password;
 
     await this.#clearKeyrings();
+    this.#keyringsMetadata = [];
```

Comment:
Any reason why this is not done inside `#clearKeyrings`? Feels like both keyrings and their metadata are intrinsically linked together, so I'd would clear the metadata in that method directly.

Reply:
We left it outside, because we are clearing the keyrings on the extension lock and we want to persist the `keyringsMetadata` unchanged in that case. So whenever user is unlocking it, metadata should be already there unchanged. 
We want to clear it here, for when the user is using forgot password flow and is setting a new vault